src/app/Library/Widget.php

```diff
@@ -133,11 +133,10 @@ public function getFinalViewPath()
                 return $path;
             }
         }
-
         $type = $this->type;
         $paths = array_map(function ($item) use ($type) {
             return $item.'.'.$type;
-        }, config('backpack.base.component_view_namespaces.widgets'));
+        }, array_merge(config('backpack.base.component_view_namespaces.widgets'), $this->crud()->get('viewNamespaces')['widgets'] ?? []));
```

Comment:
Ok so if we are to move how we get/set view namespaces, let's _do it right_. Let's add a trait to the CrudPanel object with:

```php
private $viewNamespaces;
public function getViewNamespacesFor($domain) {}; // where domain is "buttons", "columns", "fields" or "filters";
public function addViewNamespaceFor($domain, $path) {}; // where path is a blade path like "crud::fields"
```

Also:
- [ ] (maybe) let's make sure `addViewNameSpaceFor()` only adds that namespace if it doesn't already exist;

Reply:
Scratch that. See https://github.com/Laravel-Backpack/CRUD/pull/4520#discussion_r920965512 please